### ISSUE TITLE
Duplicated include with instance.reload and scope

### ISSUE THREAD

**Issue by Verdier** [sequelize/sequelize#4658]
Since sequelize `3.9.0` up to version `3.12.0`, I obtain the following error for one of  my queries: 
`[error: table name "transaction" specified more than once]`

The SQL query is:
```sql
SELECT
  "operation".*,
  ...
FROM (SELECT
        "operation"."id",
        ...
      FROM "operations" AS "operation"
      WHERE "operation"."id" = 29556
      LIMIT 1) AS "operation" LEFT OUTER JOIN "breakdowns" AS "breakdowns"
    ON "operation"."id" = "breakdowns"."operationId"
  LEFT OUTER JOIN "investments" AS "breakdowns.associationInvestment"
    ON "breakdowns"."associationInvestmentId" = "breakdowns.associationInvestment"."id"
  LEFT OUTER JOIN "cars" AS "breakdowns.associationCar"
    ON "breakdowns"."associationCarId" = "breakdowns.associationCar"."id"
  LEFT OUTER JOIN "transactions" AS "transaction" ON "operation"."transactionId" = "transaction"."id"
  LEFT OUTER JOIN "transactions" AS "transaction" ON "operation"."transactionId" = "transaction"."id";
````
The `LEFT OUTER JOIN` on `transactions` appears two times. 

I'm looking for the source a solution from a while, but I couldn't reproduce the problem using a simple configuration. I use many `scopes` on both associations and queries. Looks like there is a merge issue.

Do you have any idea, feedback, to help me find the bug?

**Comment by mickhansen** [comment]
It's most likely a scope merging issue. Try to identify all scopes that add that include, and all call sites that use that scope.

If you only have one scope that adds that include, maybe we should have a guard against double scoping the same scope @janmeier?

Guarding against double includes will be hard though.

**Comment by Verdier** [comment]
I'm now able to reproduce the issue, the problem appear when I try to reload an instance which have a `duplicate` include:
```js
var User = sequelize.define('User', {
    name: Sequelize.STRING
});

var Project = sequelize.define('Project', {
    title: Sequelize.STRING
});

User.addScope('defaultScope', {
    include: [Project]
}, {
    override: true
});

Project.belongsTo(User);
User.hasMany(Project);

let user = yield User.create({
    name: 'John'
});
yield user.createProject({
    name: 'Project 1'
});

user = yield User.findById(user.id, {
    include: [Project]
});

yield user.reload(); // Crash Here !
```

**Comment by mickhansen** [comment]
It works on find but not on reload? Odd.

**Comment by Verdier** [comment]
Yep exact @mickhansen...!

**Comment by Verdier** [comment]
And if I do something like that, I have the exact same issue:
```js
user = yield User.findById(user.id, user.$options);
```

**Comment by blackening** [comment]
I have most likely tracked down the bug:

The symptom is that :

In lib/model.js, Model.$injectScope
https://github.com/sequelize/sequelize/blob/b5bec0c64d5d91da45e699cd2b1738ecd90aedaf/lib/model.js#L2614-L2624
``` javascript
scope.include.reverse().forEach(function (scopeInclude) {
      if (scopeInclude.all || !_.any(options.include, function matchesModelAndAlias(item) {
        var sameModel = item.model && item.model.name === scopeInclude.model.name;
        //at this point, item.as is defined, but scopeInclude.as is not.
        if (sameModel && item.as) {
          return item.as === scopeInclude.as;
        }
        return sameModel;
})) ...
```
This immediately causes the sameModel to be included twice.
For example, in my case, i have:

```
item:{ model: Deal,
  parent: { /* omitted */ },
  association: Deal,
  as: 'Deal',
  attributes: [ 'id', 'contents', 'createdAt', 'updatedAt' ],
  required: false,
  duplicating: undefined,
  hasDuplicating: undefined,
  hasRequired: false,
  hasParentWhere: false,
  hasParentRequired: false,
  subQuery: false,
  subQueryFilter: false } 

scopeInclude: { model: Deal }
```


The reason for this appears to be, in
Model.prototype.findAll
Actions are done in the following order:
```
conformOptions(options, this);
Model.$injectScope(this.$scope, options);
expandIncludeAll.call(this, options); //This expands option.includes
validateIncludedElements.call(this, options, tableNames); 
| - validateIncludedElement
| | - Model.$injectScope(model.$scope, include);

```
It appears that `options.include` are provided fully expanded by reload() but nothing in here provides an expanded `scope`.

**Comment by blackening** [comment]
I have also written a small band aid solution, which does not solve the root of the problem, since that essentially involves either
a) Cleaning out the ```option.include``` from reload
b) Populating the ```scope.include```.
```
 function matchesModelAndAlias(item) {

        var sameModel = item.model && item.model.name === scopeInclude.model.name;

        var scopeIncludeAs = scopeInclude.as || scopeInclude.model.name; //Band aid solution.
        if (sameModel && item.as) {
          return item.as === scopeIncludeAs;
        }
        return sameModel;
      })
```

**Comment by janmeier** [comment]
@Verdier This one should be fixed by https://github.com/sequelize/sequelize/pull/5106 as well, right?

**Comment by Verdier** [comment]
@janmeier Yep I'm confirm that's fixed by #5106
